### ISSUE TITLE
Lenovo Yoga 7i

### ISSUE THREAD

**Issue by NovaViper** [EmixamPP/linux-enable-ir-emitter#133]
**Ouput of `linux-enable-ir-emitter -v configure`**
<details><summary>configure command output</summary>

```
❯ sudo linux-enable-ir-emitter -v configure
INFO: Ensure to not use the camera during the execution.
INFO: Warning to do not kill the process !
INFO: Configuring the camera: /dev/v4l/by-path/pci-0000:00:14.0-usb-0:6:1.2-video-index0.
Is the video flashing? Press Y or N

(generate-driver:614937): dbind-WARNING **: 13:45:15.911: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
N pressed
DEBUG: unit: 4 selector: 2
DEBUG: current: 3 0
DEBUG: minimum: 0 0
DEBUG: maximum: 255 255
DEBUG: new current: 0 0
DEBUG: new current: 1 0
DEBUG: new current: 2 0
DEBUG: new current: 3 0
DEBUG: new current: 4 0
DEBUG: new current: 5 0
DEBUG: new current: 6 0
DEBUG: new current: 7 0
DEBUG: new current: 8 0
DEBUG: new current: 9 0
DEBUG: new current: 10 0
DEBUG: new current: 11 0
DEBUG: new current: 12 0
DEBUG: new current: 13 0
DEBUG: new current: 14 0
DEBUG: new current: 15 0
DEBUG: new current: 16 0
DEBUG: new current: 17 0
DEBUG: new current: 18 0
DEBUG: new current: 19 0
DEBUG: new current: 20 0
DEBUG: new current: 21 0
DEBUG: new current: 22 0
DEBUG: new current: 23 0
DEBUG: new current: 24 0
DEBUG: new current: 25 0
DEBUG: new current: 26 0
DEBUG: new current: 27 0
DEBUG: new current: 28 0
DEBUG: new current: 29 0
DEBUG: new current: 30 0
DEBUG: new current: 31 0
DEBUG: new current: 32 0
DEBUG: new current: 33 0
DEBUG: new current: 34 0
DEBUG: new current: 35 0
DEBUG: new current: 36 0
DEBUG: new current: 37 0
DEBUG: new current: 38 0
DEBUG: new current: 39 0
DEBUG: new current: 40 0
DEBUG: new current: 255 255
ERROR: Impossible to find your infared camera
ERROR: The configuration has failed.
INFO: Do not hesitate to visit the GitHub !
INFO: https://github.com/EmixamPP/linux-enable-ir-emitter/blob/master/docs/README.md
```
</details>

**Output of `v4l2-ctl --list-devices`**
```
Integrated RGB Camera: Integrat (usb-0000:00:14.0-6)
    /dev/video0
    /dev/video1
    /dev/video2
    /dev/video3
    /dev/media0
    /dev/media1
```

**Ouput of `v4l2-ctl -d /dev/videoX --list-formats-ext`**

<details><summary>v4l2 device output</summary>

```
❯ v4l2-ctl -d /dev/video0 --list-formats-ext
ioctl: VIDIOC_ENUM_FMT
        Type: Video Capture

        [0]: 'MJPG' (Motion-JPEG, compressed)
                Size: Discrete 1920x1080
                        Interval: Discrete 0.033s (30.000 fps)
                Size: Discrete 1280x720
                        Interval: Discrete 0.033s (30.000 fps)
                Size: Discrete 960x540
                        Interval: Discrete 0.033s (30.000 fps)
                Size: Discrete 848x480
                        Interval: Discrete 0.033s (30.000 fps)
                Size: Discrete 640x480
                        Interval: Discrete 0.033s (30.000 fps)
                Size: Discrete 640x360
                        Interval: Discrete 0.033s (30.000 fps)
                Size: Discrete 352x288
                        Interval: Discrete 0.033s (30.000 fps)
                Size: Discrete 320x240
                        Interval: Discrete 0.033s (30.000 fps)
                Size: Discrete 320x180
                        Interval: Discrete 0.033s (30.000 fps)
        [1]: 'YUYV' (YUYV 4:2:2)
                Size: Discrete 1920x1080
                        Interval: Discrete 0.200s (5.000 fps)
                Size: Discrete 1280x720
                        Interval: Discrete 0.100s (10.000 fps)
                Size: Discrete 960x540
                        Interval: Discrete 0.067s (15.000 fps)
                Size: Discrete 848x480
                        Interval: Discrete 0.050s (20.000 fps)
                Size: Discrete 640x480
                        Interval: Discrete 0.033s (30.000 fps)
                Size: Discrete 640x360
                        Interval: Discrete 0.033s (30.000 fps)
                Size: Discrete 352x288
                        Interval: Discrete 0.033s (30.000 fps)
                Size: Discrete 320x240
                        Interval: Discrete 0.033s (30.000 fps)
                Size: Discrete 320x180
                        Interval: Discrete 0.033s (30.000 fps)

❯ v4l2-ctl -d /dev/video1 --list-formats-ext
ioctl: VIDIOC_ENUM_FMT
        Type: Video Capture


❯ v4l2-ctl -d /dev/video2 --list-formats-ext
ioctl: VIDIOC_ENUM_FMT
        Type: Video Capture

        [0]: 'GREY' (8-bit Greyscale)
                Size: Discrete 640x360
                        Interval: Discrete 0.067s (15.000 fps)
                        Interval: Discrete 0.033s (30.000 fps)

❯ v4l2-ctl -d /dev/video3 --list-formats-ext
ioctl: VIDIOC_ENUM_FMT
        Type: Video Capture
```
</details>

**Output of `ls -l /dev/v4l/by-path`**
```
❯ ls -l /dev/v4l/by-path
lrwxrwxrwx 81,0 root 20 Sep 13:50  pci-0000:00:14.0-usb-0:6:1.0-video-index0 -> ../../video0
lrwxrwxrwx 81,1 root 20 Sep 13:50  pci-0000:00:14.0-usb-0:6:1.0-video-index1 -> ../../video1
lrwxrwxrwx 81,2 root 20 Sep 13:50  pci-0000:00:14.0-usb-0:6:1.2-video-index0 -> ../../video2
lrwxrwxrwx 81,3 root 20 Sep 13:50  pci-0000:00:14.0-usb-0:6:1.2-video-index1 -> ../../video3
```

**Give more information if you have**
I got the camera fully configured on the very first time I ran the command (I was using ArchLinux on a live USB at the time to test the camera); and the values I had were `unit: 7, selector 6, current: 1 3 2 0 0 0 0 0 0`. After I found it was working, I booted back into NixOS only to realize that I forgot to backup the driver file the tool made (but had taken a picture of the values on my phone), so I tried to go and rerun the package (also was testing out this [nixpkg PR](https://github.com/NixOS/nixpkgs/pull/216245) since that was the only way to get this application working on NixOS).
The app would run but unlike before, the IR camera's light would turn off as it was switching to the next selector but never turn back on, and eventually the app would fail with "Impossible to find your infrared camera" in the logs. And the issue would only go away after I restarted the computer but would occur again each and every time I ran the app. I don't have the original log for when it happened but I posted what I remember it doing in the logs.

**Additional info**
 - Computer (or camera) model: Yoga 7 16IRL8 - Type 82YN
 - Linux distribution: Nixos 23.05 and EndeavourOS
 - Version of linux-enable-ir-emitter: 5.0.2

**Comment by EmixamPP** [comment]
I'm not sure to fully understand your case. 

Please use the correct issue template in order to provide all the necessary info: 
https://github.com/EmixamPP/linux-enable-ir-emitter/issues/new?assignees=&labels=&projects=&template=configuration-has-failed.md&title=

**Comment by NovaViper** [comment]
Hey sorry for the delayed response! I did reboot each time before and after I ran the commands.

However, I actually can't replicate the issue anymore; looks like adding the values I saved on my phone made it work properly again. Even when letting it run again after deleting the driver off the computer completely, the camera doesn't do that weird behavior of it just not taking anymore commands like it was doing before. I've edited my the issue so if anyone else comes across it maybe they could find something out of it. But I'm going to close it since adding the values I saved fixed the issue.
